### PR TITLE
[V2] Add github app integration check

### PR DESCRIPTION
**Pull request by ozcodes**
## 🚀 Pull Request Overview

Please provide a brief summary of your changes and the motivation behind them.

- **Related Issues/Tickets**: 
- **Type of Change** (bugfix, feature, refactor, etc.): 
- **Breaking Changes?**: 

---

### 📜 Tests Checklist

Before submitting this PR, ensure that all tests pass **and** meet the following conditions:

| Category                  | Status                          | Compilation Comment | Results Review |
|---------------------------|----------------------------------|----------------------|----------------|
| **fail_to_start**         | ❌ Failed with compilation error | 🟢 Yes               | 🔴 No          |
| **violated_rules**        | ✅ Passed, with violations found | 🔴 No                | 🟢 Yes         |
| **verified_rules**        | ✅ Passed without violations     | 🔴 No                | 🟢 Yes         |
| **solana_violated_rules** | ✅ Passed, with violations found | 🔴 No                | 🟢 Yes         |

> Please verify that your changes meet the above conditions for each category of tests.  
> If something doesn't match, investigate before submitting the PR.

---

### 📋 Additional Notes (Optional)

- Any risks, edge cases, or implementation notes for reviewers:

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -279,8 +279,11 @@ runs:
       shell: bash
       run: |
         RUN_CERTORA="${{ github.action_path }}/scripts/run-certora.sh"
+        GH_APP_INTEGRATION="${{ github.action_path }}/scripts/gh-app-integration.sh"
+
         if [[ "$CERTORA_COMPILATION_STEPS_ONLY" == "false" ]]; then
           CERTORA_COMPILATION_STEPS_ONLY=true bash "$RUN_CERTORA" && \
+          "$GH_APP_INTEGRATION" && \
```
[the remark this answers is not in the document]
Comment (ozcodes) [reply action.yml:286]: no its a regular bash script, it worked for me without it but i'll add it just in case

File: scripts/gh-app-integration.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    DATA_API="data-api-stg"
+    ;;
+  vaas-dev)
+    DATA_API="data-api-dev"
+    ;;
+  *)
+    echo "::error title=Invalid Certora Server::Certora Server must be set to 'production', 'staging', or 'vaas-dev'."
+    exit 1
+    ;;
+esac
```
[the remark this answers is not in the document]
Comment (ozcodes) [reply scripts/gh-app-integration.sh:46]: than should we change CERTORA_SERVER options in the action.yml ? currently its set to production, staging and vaas-dev

[the remark this answers is not in the document]
Comment (ozcodes) [reply scripts/gh-app-integration.sh]: changed
